### ISSUE TITLE
Don't apply taxes to tickets by default

### ISSUE THREAD

**Issue by Pebblo** [eventespresso/event-espresso-core#3047]
Taxes are applied to tickets by default, so even if you leave the price as 0.00 and publish you'll see the `*` and `* price includes taxes` on the ticket selector.

To reproduce, create a new event, set the title and datetime title and ticket title as you would.

Then view the event.

The legacy editor has taxes disabled by default so I'm guessing those shouldn't be applied here.

Also, should free tickets ever show as being taxable?

**Comment by tn3rb** [comment]
@manzoorwanijk 

can you please look into this?

we want to ensure the following business rules are always applied:

- taxes are not added by default, the user MUST explicitly click to "add taxes" button to do so

- ONLY tickets that have taxes added have their `isTaxable` flag toggled to `true`

- removing taxes from a ticket should set the  `isTaxable` flag toggled to `false`

**Comment by manzoorwanijk** [comment]
I think it may be the case of default entities where the default ticket is added with default taxes? 🤔

**Comment by manzoorwanijk** [comment]
After investigating into the issue. It seems like those default prices are added in
`core/domain/services/admin/entities/DefaultTickets.php`

**Comment by tn3rb** [comment]
hmmmm... here's the code in `DefaultPrices` that loops thru default prices and applies them:

```php
    if ($default_price->is_tax()) {
        $entity->set_taxable(true);
        $default_price->_add_relation_to($entity, 'Ticket');
        continue;
    }
```

so it is only toggling the taxable flag if a default price is a tax.

need to investigate further

**Comment by tn3rb** [comment]
@Pebblo 

I've investigated this issue more and can not recreate the issue via the steps that you have implied:

> To reproduce, create a new event, set the title and datetime title and ticket title as you would.

You're not clear as to what exactly you did during the ticket creation process, but if you simply create a ticket and skip the ticket prices step, or do not add default prices, then the ticket is not marked as taxable. The only way I can recreate this issue is if I add a tax to a ticket but do NOT set a base price nor any fixed surcharges, which effectively adds whatever % of tax you applied to a $0 ticket, resulting in a $free ticket.

> Taxes are applied to tickets by default, so even if you leave the price as 0.00 and publish you'll see the * and * price includes taxes on the ticket selector. 

this is not exactly correct. You have to **manually** add default prices. When you first create a brand new ticket and advance to the ticket prices form step, you should see the following:

![ScreenShot_20200803120109](https://user-images.githubusercontent.com/1751030/89217478-1cdd8b00-d581-11ea-8c99-5abdce282d7c.png)


If you click the "Add default prices" button and you have a tax setup as a default price, then that will be included in the price modifiers displayed in the Ticket Price Calculator (TPC).
ex:

![ScreenShot_20200803115834](https://user-images.githubusercontent.com/1751030/89217232-ae003200-d580-11ea-8071-78cd6ece106b.png)


But as you can see, the TPC is warning you that you have not set a base price via the red border around that field. We have not included any further validation and/or prevent the user from submitting the form in this state, although maybe that is something we should do.

Bottom line, it appears that the EDTR and TPC have done exactly what you told it to do, unless there are some other steps you took to cause the issue, in which case you will have to be more clear and explicit about how to recreate the problem.

**Comment by Pebblo** [comment]
> You're not clear as to what exactly you did during the ticket creation process

Yep, that's because, I'm not creating the ticket, EE is. Sorry if this isn't clear but these instructions:

> To reproduce, create a new event, set the title and datetime title and ticket title as you would.

Are actually the exact steps I'm taking to reproduce this.

Create a new event (the only change I've made is to put something in the event title), look in the Event Dates and Available Tickets Section. There's a Datetime and a Ticket created automagically:
https://monosnap.com/file/MhOrNrFyxyxbaStLycrUIdt7Zj4ziq

Edit the Datetime title and the Ticket Title:
https://monosnap.com/file/cnsPUAgz8Z6a4kOqzMimDEOBlNebnh

Hit publish.

View the event: https://monosnap.com/file/vjjUSv2nJ7wViaTRzKiVx8dwqFiUTe

I've not set default tickets, default pricing or anything of the sort on this site, it's only been used for editor testing so far and was a clean install so anything set was set by EE.

> this is not exactly correct. You have to manually add default prices. When you first create a brand new ticket and advance to the ticket prices form step, you should see the following:

Ok, but that's not what is happening for me with the above. Taxes are being applied by default to the ticket EE creates by default. I'm not creating tickets, setting a price, or using the TPC for any of this, I'm simply editing the Datetime and Ticket titles for the default ticket.

If I'm a user creating a free event, those are the exact steps I would take (plus some additional event content). I wouldn't load the TPC for a free ticket, nor would I set a price as the default one is already free.

> Bottom line, it appears that the EDTR and TPC have done exactly what you told it to do, unless there are some other steps you took to cause the issue, in which case you will have to be more clear and explicit about how to recreate the problem.

I haven't told TPC to do anything which is why I haven't included the details for it here, the above is literally exactly what I'm doing to recreate but I'm guessing there's something different with my setup here?

**Comment by tn3rb** [comment]
OK so I tested this again just using whatever default ticket was created, and the system still essentially did exactly what it was told to do, and the only difference is that the free tickets were not flagged as being taxable EVEN if they had taxes applied to them. So the ticket selector did not tag the tickets with an asterisk and display the "prices includes taxes" notice.

so basically if you:

- created a default ticket with a $0 base price

- created system taxes and added those as default price modifiers

then when creating a new event, a free ticket with a tax applied to it would be created, but like i mentioned above, the ticket would not be flagged as taxable. I noticed in your screenshots that you were using an older version of the EDTR, so that likely explains the differences we were seeing.

### ANYWAYS...

that all said, I've added some user proofing logic to the default price creation logic, that will ONLY apply default taxes if the ticket has a non-zero price. This is essentially NOT doing what you are asking the system to do, but seems to be a sensible safeguard to keep users from shooting themselves in the foot.

I'll get this change from #3069 merged into the `EDTR/production` branch so that you can retest things.

**Comment by tn3rb** [comment]
this is fixed in the `barista-prod` branch now

**Comment by garthkoyle** [comment]
Can this be tested on brent.eventespresso.com now?

**Comment by garthkoyle** [comment]
@Pebblo can you test this please?

**Comment by tn3rb** [comment]
> Can this be tested on brent.eventespresso.com now?

yes the the `barista-prod` branch is active on `brent.eventespresso.com`

**Comment by Pebblo** [comment]
This is now fixed in Barista.

With a new, clean install just create an event add an event title and publish.

Theres a free ticket that no longer shows its taxable.
